### DISCUSSION TITLE
expected motif enrichment has footprints

### DISCUSSION THREAD

**Discussion by Connorr0** [stuart-lab/signac#717] [Q&A] [answered]
I was doing some reading on the footprinting and had some questions.

In my expected motif enrichment, I noticed I was seeing a "footprint" in one of the motif sequences I was interested in:

![image](https://user-images.githubusercontent.com/29585780/125682296-ab8ba129-9161-431b-8176-692cb05c4fd5.png)

I was looking at your biorxiv paper for Signac, and was reading up on the expected motif enrichment scores:

>To account for Tn5 sequence insertion bias, we first computed the observed Tn5 insertion frequency at each DNA hexamer using all Tn5 insertions on chromosome 1. This was done by extracting the base-resolution Tn5 insertion positions for each fragment mapped to chromosome 1, and extending the insertion coordinate 3 bp upstream and 2 bp downstream. We then extracted the DNA sequence corresponding to these coordinates using the getSeq function from the Biostrings R package [Pagès et al., 2020] and counted the frequency of each hexamer using the table function in R. We next computed the expected Tn5 hexamer insertion frequencies based on the frequency of each hexamer on chromosome 1. We counted the frequency of each hexamer using the oligonucleotideFrequency function in the Biostrings package with width=6 and names=“chr1”, using the hg38 genome via the BSgenome R package [Pagès, 2020]. Finally, we computed the Tn5 insertion bias as the observed Tn5 insertions divided by the expected insertions at each hexamer. This was performed using the InsertionBias function in Signac.

At this time the factor I am interested in (one of the ONECUTs) is expressed (and is thought to be active) in most cells in my sample. I thought that this could be what was causing the "footprint" in the expected motif sequence. Since the bias is computed across all cells(?) on chr1, and since the motifs/binding sites could be hitting a cluster of hexamers, I am worried that my expected motif enrichment is assuming that there is footprinting activity, rather than just correcting for TN5 bias.

I was wondering if this was something you have considered and had been working on counteracting.

Thanks for the awesome tool.

**Comment by timoast** [comment, marked as answer]
> Since the bias is computed across all cells(?) on chr1, and since the motifs/binding sites could be hitting a cluster of hexamers, I am worried that my expected motif enrichment is assuming that there is footprinting activity, rather than just correcting for TN5 bias.

If there was a very extreme case where **every** instance (across all of chr1) of a certain hexamer was blocked by the strong binding of a protein in **all cells,** then this would appear to be a strong bias against Tn5 integration at that hexamer which we assume here is an inherent Tn5 sequence preference, but could (in this hypothetical) be a biological phenomenon. However, it's very unlikely that you'd see this in practice.

It's difficult to conclude that there's really "footprinting" in the scATAC-seq data, due to the Tn5 insertion bias (I used this term in the paper because it's so established in the field, but I don't think you're really measuring footprinting per se). What is more useful to look at is the accessibility at the motif flanks. If there is high accessibility flanking the motif sites, then there's good evidence that those sites are open/active in that particular cell type (which may be driven by TF binding at those sites). I would basically ignore the sequence overlapping the motif itself, and look only at the motif flanks.

**Reply by dmiyagi** [reply]
I am wondering if this might be a case for what you are referring to? I am trying to figure out why the Tn5 enrichment is completely blank at the footprint. I am studying a tumor with rather odd characteristics that may be being driven by epigenetic factors, so I am wondering if I can interpret this as there being strong binding across all cells as you mentioned.  <img width="561" alt="Screen Shot 2022-06-11 at 12 31 51 PM" src="https://user-images.githubusercontent.com/47726144/173196602-77cd1c4b-55c1-49ef-a791-4a4e350f7b45.png">

notably, this is not the case for all the motifs in my sample, but one oddity is that this particular example is actually the motif expected of a different species which I am not aligning to so I'm not sure how/why it comes up? Thanks in advance

<img width="709" alt="Screen Shot 2022-06-11 at 12 35 03 PM" src="https://user-images.githubusercontent.com/47726144/173196725-eb2c7056-8a9b-4dc3-acc9-5f20c3843500.png">
